src/middlewared/middlewared/plugins/virt/volume.py

```diff
@@ -46,6 +48,11 @@ async def query(self, filters, options):
     async def do_create(self, data):
         await self.middleware.call('virt.global.check_initialized')
 
+        verrors = ValidationErrors()
+        if await self.middleware.call('virt.volume.query', [['id', '=', data['name']]]):
+            verrors.add('virt_volume_create.name', 'Volume with this name already exists')
```

Comment:
Just import ValidationError and raise it here instead of creating a verrors object